djangocms_versioning_filer/monkeypatch/admin/admin_fields.py

```diff
@@ -0,0 +1,184 @@
+from django.urls import reverse
+from django.utils.http import urlencode
+from django.utils.safestring import mark_safe
+
+from cms.models.pluginmodel import CMSPlugin
+from cms.models.titlemodels import PageContent
+
+import filer
+from djangocms_versioning.constants import DRAFT, OPERATION_PUBLISH
+from djangocms_versioning.models import Version
+from djangocms_versioning.signals import post_version_operation
+from filer import settings as filer_settings
+from filer.fields.file import AdminFileFormField, FilerFileField
+from filer.models.filemodels import File
+
+
+logger = logging.getLogger(__name__)
+
+
+def render(self, name, value, attrs=None, renderer=None):
+    obj = self.obj_for_value(value)
+    css_id = attrs.get('id', 'id_image_x')
+    related_url = None
+    if value:
+        try:
+            # We need to get the original manager to be able to access drafts.
+            file_obj = File._original_manager.get(pk=value)
+            related_url = file_obj.logical_folder.get_admin_directory_listing_url_path()
+        except Exception as e:
+            # catch exception and manage it. We can re-raise it for debugging
+            # purposes and/or just logging it, provided user configured
+            # proper logging configuration
+            if filer_settings.FILER_ENABLE_LOGGING:
+                logger.error('Error while rendering file widget: %s', e)
+            if filer_settings.FILER_DEBUG:
+                raise
+    if not related_url:
+        related_url = reverse('admin:filer-directory_listing-last')
+    params = self.url_parameters()
+    params['_pick'] = 'file'
+    if params:
+        lookup_url = '?' + urlencode(sorted(params.items()))
+    else:
+        lookup_url = ''
+    if 'class' not in attrs:
+        # The JavaScript looks for this hook.
+        attrs['class'] = 'vForeignKeyRawIdAdminField'
+    # rendering the super for ForeignKeyRawIdWidget on purpose here because
+    # we only need the input and none of the other stuff that
+    # ForeignKeyRawIdWidget adds
+    hidden_input = super(ForeignKeyRawIdWidget, self).render(name, value, attrs)
+    context = {
+        'hidden_input': hidden_input,
+        'lookup_url': '%s%s' % (related_url, lookup_url),
+        'object': obj,
+        'lookup_name': name,
+        'id': css_id,
+        'admin_icon_delete': ('admin/img/icon-deletelink.svg'),
+    }
+    html = render_to_string('admin/filer/widgets/admin_file.html', context)
+    return mark_safe(html)
+
+
+def obj_for_value(self, value):
+    if value:
+        try:
+            key = self.rel.get_related_field().name
+            # we need the original manager to access draft
+            obj = self.rel.model._original_manager.get(**{key: value})
+        except Exception:
+            obj = None
+    else:
+        obj = None
+    return obj
+
+
+def __init__(self, rel, queryset, to_field_name, *args, **kwargs):
+    def remove_published_where(queryset):
```

Comment:
This is a helper method that could be added to helpers.py, it could also be added to versioning later as a helper too.

The init method doesn't need to repeat the logic, it can be written in a way to only make the change required to the queryset. It's a shame the other methods aren't written better so that we could have patched them in a cleaner way.

```

# Helpers.py 
def remove_published_where(.........


def __init__(func):
    def inner(self, rel, queryset, to_field_name, *args, **kwargs):
        queryset = remove_published_where(queryset)
        func(self, rel, queryset, to_field_name, *args, **kwargs)
    return inner
```

Reply:
or rather should work..  

```
def __init__(self, rel, queryset, to_field_name, *args, **kwargs):
    queryset = remove_published_where(queryset)
    super(AdminFileFormField, self).__init__(queryset, widget=self.widget(rel, site), *args, **kwargs)
```

Reply:
This is now fixed.